### PR TITLE
Unifying boot, by using partition label resin-rootA/B instead of physical mmcblk0p2/3

### PR DESCRIPTION
**Pull request by theonlyzby**
Partition label to boot instead of physical one, in order to boot from SD Card, USB or HardDisk.
Cf Forum discussion with @agherzan 
https://forums.resin.io/t/raspberry-to-run-resin-io-from-hard-disk/2413
https://github.com/resin-os/meta-resin/issues/918

I hope this will help, Henry

### CODE REVIEW

File: layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend [hunk 1]
```diff
@@ -9,8 +9,8 @@ LINUX_VERSION = "4.9.59"
 SRCREV = "e7976b2affa8c05bd299ead0ad9c36ddaf135b9d"
 
 # Set console accordingly to build type
-DEBUG_CMDLINE = "dwc_otg.lpm_enable=0 console=tty1 console=serial0,115200 root=/dev/mmcblk0p2 rootfstype=ext4 rootwait"
-PRODUCTION_CMDLINE = "dwc_otg.lpm_enable=0 console=null root=/dev/mmcblk0p2 rootfstype=ext4 rootwait vt.global_cursor_default=0"
+DEBUG_CMDLINE = "dwc_otg.lpm_enable=0 console=tty1 console=serial0,115200 root=LABEL=resin-rootA rootdelay=5 fsck.repair=yes rootfstype=ext4 rootwait"
+PRODUCTION_CMDLINE = "dwc_otg.lpm_enable=0 console=null root=LABEL=resin-rootA rootdelay=5 fsck.repair=yes rootfstype=ext4 rootwait vt.global_cursor_default=0"
```
Comment (floion) [inline layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend]: Can you leave out of this PR the addition of "fsck.repair=yes" and "rootdelay=5" ?

Comment (floion) [reply layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend]: Please change the commit log to something like:

> linux-raspberrypi_%.bbappend: Use filesystem labels for the root cmdline kernel parameter

Followed by a detailed description on why this change is necessary

Comment (floion) [reply layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend]: Also, please add a CHANGELOG.md entry with a short line for this change, like in: 0ab6e8d

**Comment by floion** [comment]
@resin-jenkins ok to test

**Comment by floion** [comment]
Can you squash the 2 commits into 1?
And also add another commit with the CHANGELOG entry please?

**Comment by theonlyzby** [comment]
I realize I should have done 1 commit for both changes, but I am a GIT newbee ... sorry still don't know how to fix this! And Changelog should be OK, as I submitted both modifications into it

File: layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend (hunk 1 above)
Comment (floion) [reply layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend]: Please redo this commit.
You need to remove the
> rootdelay=5 fsck.repair=yes

additions from this commit. After you do that, you can drop the other commit: https://github.com/resin-os/resin-raspberrypi/pull/165/commits/63b71ef42284670d90ceb3ed4e64a0d967eeb1c6

To do this, in your branch you should use "git rebase".

File: layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend [hunk 2]
```diff
@@ -14,6 +14,7 @@ PRODUCTION_CMDLINE = "dwc_otg.lpm_enable=0 console=null root=LABEL=resin-rootA r
 CMDLINE = "${@bb.utils.contains('DISTRO_FEATURES','development-image',"${DEBUG_CMDLINE}","${PRODUCTION_CMDLINE}",d)}"
 CMDLINE_DEBUG = ""
 
+
```
Comment (floion) [inline layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend]: Hi. This PR is still wrong.
Please see my suggestions from the other thread

Comment (theonlyzby) [reply layers/meta-resin-raspberrypi/recipes-kernel/linux/linux-raspberrypi_%.bbappend]: Florin, 
I am sorry, maybe too newbee on GitHub, but I can't decode what you ask me to do... It seems to me I removed the HDD specific arguments and ONLY changed the mount point, no? 
Henry

**Comment by floion** [comment]
I think maybe you can start over with a new PR?
Unless you can squash these commits in the correct form: https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by theonlyzby** [comment]
I will try to do it from Desktop app, not from web.

**Pull request closed without merging**